### PR TITLE
Dev v5 fast sim btagging

### PR DESCRIPTION
**Pull request by scarletnorberg**

### CODE REVIEW

**Comment by pastika** [comment]
the PR looks good to me

**Comment by scarletnorberg** [comment]
Can this be merged then?

File: python/processors/Stop0l_postproc.py [hunk 1]
```diff
@@ -278,9 +279,9 @@ def main(args):
                           photonSelectionTag="Medium", 
                           tauSelectionTag="Tight"),
             puWeightProducer(pufile_mc, pufile_data, args.sampleName,"pileup"),
-            # btagSFProducer(era=BTagERA, algo="deepcsv"),
+            btagSFProducer(args.era, algo="deepcsv"),
             #BtagSFWeightProducer("allInOne_bTagEff_deepCSVb_med.root", args.sampleName, DeepCSVMediumWP[args.era]),
-            # BtagSFWeightProducer(DataDepInputs[dataType][args.era]["bTagEff"], args.sampleName, DeepCSVMediumWP[args.era]),
+            BtagSFWeightProducer(DataDepInputs[dataType][args.era]["bTagEff"], args.sampleName, DeepCSVMediumWP[args.era], isfastsim),
```
Comment (zhenbinwu) [inline python/processors/Stop0l_postproc.py]: args.sampleName won't work for fastsim. Take a  look at the ISR example

**Comment by zhenbinwu** [comment]
Ok, let me add more information to the above request:

We will have many many signal points. To handle this, we spited the pre-process file per signal points. To avoid a very long cfg file, we will use the file name as the sample name. 

Since the processing time of signal point isn't very long, it is better to submit job with multiple signals per job.

In the Stop0l_postproc, the args.samplename is used in the modules , which was defined per job. However, we need to get this information per file. We can take the advantage of the beginFile() in the modules, which is called per input root file.

That is why we can follow step like https://github.com/susy2015/NanoSUSY-tools/blob/dev_v5_fast/python/modules/ISRWeightProducer.py#L79 as a solution.

Hope to make it clear

**Comment by zhenbinwu** [comment]
Looks fine. Merging this PR

**Pull request merged**